src/components/draw/DrawStyleDirective.js

```diff
@@ -15,16 +15,46 @@ goog.require('ga_window_service');
   module.directive('gaDrawStyle', function($document, $window, $translate,
       gaGlobalOptions, gaStyleFactory, gaUrlUtils, gaWindow) {
 
+    var findCategoryBySource = function(src, categories) {
+      for (var i = 0; i < categories.length; i++) {
+        var c = categories[i];
+        var regex = (c.type === 'img') ?
+          new RegExp('^(.*)' + c.id) :
+          new RegExp(c.regex);
+        if (regex.test(src)) {
+          return c;
+        }
+      }
+      console.error('No category found for source ' + src);
+    };
+
+    var getCategoryById = function(id, categories) {
+      for (var i = 0; i < categories.length; i++) {
+        var c = categories[i];
+        for (var j = 0; j < c.icons.length; j++) {
+          var icon = c.icons[j];
+          if (icon.id === id) {
+            return c;
+          }
+        }
+      }
+      console.error('No category found for id ' + id);
+    }
+
     // Find the corresponding style
-    var findIcon = function(olIcon, icons) {
+    var findIcon = function(olIcon, category, standardIcon) {
       var id = olIcon.getSrc();
+      var icons = category.icons;
       for (var i = 0; i < icons.length; i++) {
-        var regex = new RegExp('/' + icons[i].id + '-24');
+        var icon = icons[i];
+        var regex = (category.type === 'img') ?
+          new RegExp(category.id + '-' + icon.id + '.png') :
+          new RegExp(category.regex);
```

Comment:
If regex property is already a RegeEx object you don't have to recreate the regex each time.